src/order/compactly_generated.lean

```diff
@@ -470,8 +473,23 @@ theorem is_complemented_of_Sup_atoms_eq_top (h : Sup {a : α | is_atom a} = ⊤)
         refine hc2.directed_on.mono (λ s t, Sup_le_Sup) } },
     { rcases set.mem_sUnion.1 ha with ⟨s, sc, as⟩,
       exact (hc1 sc).2.2 a as } }
+end
+
+/-- See Theorem 6.6, Călugăreanu -/
+lemma is_complemented_of_Sup_atoms_eq_top
+  (h : Sup {a : α | is_atom a} = ⊤) : is_complemented α :=
+⟨λ b, begin
+  obtain ⟨s, s_ind, s_top, s_atoms⟩ := exists_set_independent_Sup_atoms_eq_compl h b,
+  refine ⟨Sup s, s_top⟩
 end⟩
 
+lemma is_atomistic.exist_set_independent_Sup_eq_top (h : Sup {a : α | is_atom a} = ⊤) :
+  ∃ s : set α, complete_lattice.set_independent s ∧ Sup s = ⊤ ∧ ∀ a ∈ s, is_atom a :=
+begin
+  obtain ⟨s, s_ind, s_top, s_atoms⟩ := exists_set_independent_Sup_atoms_eq_compl h (⊥ : α),
+  refine ⟨s, s_ind, eq_top_of_is_compl_bot (is_compl.symm s_top), s_atoms⟩,
+end
+
```

Comment:
@acxxa, do you use the new lemmas in this file in the rest of the PR? If not, can you split this out into its own PR?

Reply:
I've split it out in #8475